src/utils/exceptionThrowers.ts

```diff
@@ -0,0 +1,275 @@
+    console.log(exceptionDetails.message);
+    throw Error(exceptionDetails.response);
+};
+
+export const throwConversationInitializationFailure = (e: unknown, scenarioMarker: ScenarioMarker, telemetryEvent: TelemetryEvent, telemetryData: {[key: string]: string}): void => {
+    const exceptionDetails: ChatSDKExceptionDetails = {
+        response: ChatSDKErrors.ConversationInitializationFailure,
+        errorObject: `${e}`
+    };
+
+    scenarioMarker.failScenario(telemetryEvent, {
+        ...telemetryData,
+        ExceptionDetails: JSON.stringify(exceptionDetails)
+    });
+
+    throw Error(exceptionDetails.response);
+};
+
+export const throwConversationClosureFailure = (e: unknown, scenarioMarker: ScenarioMarker, telemetryEvent: TelemetryEvent, telemetryData: {[key: string]: string}): void => {
+    const exceptionDetails: ChatSDKExceptionDetails = {
+        response: ChatSDKErrors.ConversationClosureFailure,
+        errorObject: `${e}`
+    };
+
+    scenarioMarker.failScenario(telemetryEvent, {
+        ...telemetryData,
+        ExceptionDetails: JSON.stringify(exceptionDetails)
+    });
+
+    throw Error(exceptionDetails.response);
+};
+
+export const throwMessagingClientInitializationFailure = (e: unknown, scenarioMarker: ScenarioMarker, telemetryEvent: TelemetryEvent, telemetryData: {[key: string]: string}): void => {
+    const exceptionDetails: ChatSDKExceptionDetails = {
+        response: ChatSDKErrors.MessagingClientInitializationFailure,
+        errorObject: `${e}`
+    };
+
+    scenarioMarker.failScenario(telemetryEvent, {
+        ...telemetryData,
+        ExceptionDetails: JSON.stringify(exceptionDetails)
+    });
+
+    throw Error(exceptionDetails.response);
+};
+
+export const throwMessagingClientConversationJoinFailure = (e: unknown, scenarioMarker: ScenarioMarker, telemetryEvent: TelemetryEvent, telemetryData: {[key: string]: string}): void => {
+    const exceptionDetails: ChatSDKExceptionDetails = {
+        response: ChatSDKErrors.MessagingClientConversationJoinFailure,
+        errorObject: `${e}`
+    };
+
+    scenarioMarker.failScenario(telemetryEvent, {
+        ...telemetryData,
+        ExceptionDetails: JSON.stringify(exceptionDetails)
+    });
+
+    throw Error(exceptionDetails.response);
+};
+
+export default {
+    throwUnsupportedPlatform,
+    throwFeatureDisabled,
+    throwOmnichannelClientInitializationFailure,
+    throwUnsupportedLiveChatVersionFailure,
+    throwChatConfigRetrievalFailure,
+    throwMessagingClientCreationFailure,
+    throwUninitializedChatSDK,
+    throwChatTokenRetrievalFailure,
+    throwInvalidConversation,
+    throwClosedConversation,
+    throwAuthenticatedChatConversationRetrievalFailure,
+    throwPersistentChatConversationRetrievalFailure,
+    throwWidgetUseOutsideOperatingHour,
+    throwConversationInitializationFailure,
+    throwConversationClosureFailure,
+    throwMessagingClientInitializationFailure,
+    throwMessagingClientConversationJoinFailure
+}
```

Comment:
Where are these exceptionThrowers being used?

Reply:
They're being used in `src/OmnichannelChatSDK.ts`